docs/entityoperator.md

```diff
@@ -0,0 +1,234 @@
+props.put("security.protocol", "SSL");
+props.put("ssl.truststore.location", "/etc/secrets/ca.p12");
+props.put("ssl.truststore.password", "password");
+props.put("ssl.keystore.location", "/etc/secrets/user.p12");
+props.put("ssl.keystore.password", "password");
+
+KafkaProducer<String, String> producer = new KafkaProducer<>(props);
+```
+
+#### Java Consumer Example (SCRAM-SHA Authentication)
+
+```java
+Properties props = new Properties();
+props.put("bootstrap.servers", "my-cluster-kafka-bootstrap:9093");
+props.put("security.protocol", "SASL_SSL");
+props.put("sasl.mechanism", "SCRAM-SHA-512");
+props.put("sasl.jaas.config", "org.apache.kafka.common.security.scram.ScramLoginModule required username='my-app-user' password='my-secret-password';");
+
+KafkaConsumer<String, String> consumer = new KafkaConsumer<>(props);
+
+```
+
+### Summary of How Applications Use KafkaTopic & KafkaUser CRs
+
+| Action    | Operator Responsible |
+| -------- | ------- |
+| Developer creates a `KafkaTopic` CR  | Topic Operator creates & syncs the topic in Kafka    |
+| Developer creates a KafkaUser CR | User Operator creates the user & credentials  |
+| Application retrieves credentials from Kubernetes Secrets	| Application mounts the secrets for authentication |
+| Application connects to Kafka using these credentials | Producer/Consumer communicates with Kafka  |
+
+
+## Impact of Central Cluster Failure on Kafka Clients in a Stretch Cluster
+
+In stretch Kafka deployment, where
+
+- ✅ Kafka brokers and controllers are spread across multiple Kubernetes clusters.
+- ✅ The central cluster hosts all Kafka CRs, including Kafka, KafkaNodePool, KafkaUser, and KafkaTopic.
+- ✅ The Entity Operator (managing users & topics) runs in the central cluster.
+
+## Does the Kafka Cluster Still Function?
+
+Yes, but with conditions.
+
+- Brokers in other Kubernetes clusters are still running.
+- Kafka controllers may still be running (if a quorum is maintained).
+
+Replication between brokers (spread across clusters) continues as long as there’s a majority quorum of controllers.
+
+- ✅ If a majority of controllers are still available in the surviving clusters, Kafka continues running.
+- 🚨 If the majority of controllers were in the central cluster and lost, Kafka will experience a complete outage.
+
+💡 Mitigation -> user need to ensure the `controller.quorum.voters` are spread across clusters to avoid losing the majority.
+
+## Can Kafka Clients Still Communicate with Brokers?
+
+Yes, if at least one broker remains reachable.
+
+Clients connect to brokers, not the Entity Operator or the central cluster.
+
+- If clients' bootstrap servers list includes brokers running in the surviving clusters, they can still connect.
+- If a leader partition for a topic is hosted on a broker in a surviving cluster, the topic remains available.
+- If all leader partitions for a topic were on brokers in the central cluster, that topic becomes unavailable.
```

Comment:
In this situation Kafka should select a new leader for the topic, we should make that clear. Also, this is how Kafka works when deplyed in a single cluster and a broker goes down. So IMO this is again not specific to the entity operator or stretch cluster.